Designate the lightrec mirror entries

GCC 16 rejects mixing designated and non-designated initializers in
C++ ("either all initializer clauses should be designated or none").

--- a/mednafen/psx/cpu.cpp	2026-09-29 13:50:33.210097842 +0200
+++ b/mednafen/psx/cpu.cpp	2026-09-29 13:50:33.227556501 +0200
@@ -3689,21 +3689,21 @@
 	},
 
 	/* Mirrors of the kernel/user memory */
-	{
+	[PSX_MAP_MIRROR1] = {
 		.pc = 0x00200000,
 		.length = 0x200000,
 		.address = NULL,
 		.ops = NULL,
 		.mirror_of = &lightrec_map[PSX_MAP_KERNEL_USER_RAM],
 	},
-	{
+	[PSX_MAP_MIRROR2] = {
 		.pc = 0x00400000,
 		.length = 0x200000,
 		.address = NULL,
 		.ops = NULL,
 		.mirror_of = &lightrec_map[PSX_MAP_KERNEL_USER_RAM],
 	},
-	{
+	[PSX_MAP_MIRROR3] = {
 		.pc = 0x00600000,
 		.length = 0x200000,
 		.address = NULL,
